### PR TITLE
Added voice input for caption and description

### LINKED ISSUE #5404: Voice input for caption and description
The Wikipedia app shows voice input buttons when adding an image to an article:

Even though other apps don't show any such speech-to-text UI. So I believe the Wikipedia app added this. For privacy reasons I am generally wary of using speech-to-text, but I use and like these buttons on the Wikipedia app because I trust Wikimedia's privacy policies.

I think it would be great to offer the same in our caption&description activity within the Upload Wizard.

To avoid privacy issues, we should implement it like the Wikipedia app implements it.

### PR DESCRIPTION
**Pull request by kanahia1**
**Description (required)**

Fixes #5404

**Tests performed (required)**

Tested betaDebug on Samsung S21 FE with API level 33.

**Screenshots (for UI changes only)**

 `UploadActivity`

https://github.com/commons-app/apps-android-commons/assets/114223204/1e998caf-07fe-4fa4-bbb3-e2d61046968d

`DescriptionEditActivity`

https://github.com/commons-app/apps-android-commons/assets/114223204/58aaf6df-4d18-4fb8-8387-6086f8c0e8cb

---
Happy New Year ✨, Wikimedia Community
---

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -784,6 +784,8 @@ Upload your first media by tapping on the add button.</string>
   <string name="storage_permissions_denied">Storage Permissions Denied</string>
   <string name="unable_to_share_upload_item">Unable to share this item</string>
   <string name="permissions_are_required_for_functionality">Permissions are required for functionality</string>
+  <string name="learn_how_to_write_better_description">Learn how to write better description</string>
```
Comment (nicolas-raoul) [inline app/src/main/res/values/strings.xml:787]: I would suggest: `Learn how to write a useful description`

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -784,6 +784,8 @@ Upload your first media by tapping on the add button.</string>
...
+  <string name="learn_how_to_write_better_caption">Learn how to write better caption</string>
```
Comment (nicolas-raoul) [inline app/src/main/res/values/strings.xml:788]: I would suggest: `Learn how to write a useful caption`

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
Installed on my phone, I will try it as soon as possible. :-)

File: app/src/main/res/values/strings.xml (hunk 1 above)
Comment (kanahia1) [reply app/src/main/res/values/strings.xml]: Thank you for suggestion, I made the suggested changes in commit https://github.com/commons-app/apps-android-commons/pull/5415/commits/0dc6b1123353005fbcc881d6c89d6abee0d008a0

**Comment by nicolas-raoul** [comment]
In order to fix the unit tests, would you mind rebasing (or pulling) from main? Thanks :-)

**Comment by kanahia1** [comment]
Sure :-)

**Comment by kanahia1** [comment]
@nicolas-raoul , pulling from main has fixed the unit test issues 🙂

**Review by nicolas-raoul — APPROVED** [review]
Working great, thanks!

**Pull request merged**